include/zephyr/sys/arch_interface.h

```diff
@@ -1220,10 +1220,29 @@ bool arch_pcie_msi_vector_connect(msi_vector_t *vector,
  */
 void arch_spin_relax(void);
 
+/*
+ * @brief Check if an address is a valid physical address.
+ *
+ * @param phys Physical address to be checked.
+ *
+ * @return True if physical address is valid, false if not.
+ */
+bool arch_mem_is_valid_phys_addr(uintptr_t phys);
+
+/**
+ * @brief Check if an address is a valid virtual address.
```

Comment:
Can we be stricter with the docs?  "Valid" here means something like "can be theoretically mapped".  It's easy for someone to assume this means "is actually mapped in the currently-active page tables".

Reply:
I renamed these to mean "within range" instead of simply "being valid". Hopefully the wording is less confusing.